### PR TITLE
Add deprecation warning for GET/POST/PUT cache

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Adds deprecation warning that GET/POST/PUT methods for cache endpoint is removed in version 3.0.0. 

### Issues Resolved
None
Is this a backport? If so, please add backport PR # and/or commits #, and remove `backport-failed` label from the original PR.
No

Do these changes introduce new permission(s) to be displayed in the static dropdown on the front-end? If so, please open a draft PR in the security dashboards plugin and link the draft PR here
No

### Testing
None
### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] New Roles/Permissions have a corresponding security dashboards plugin PR
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md)
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by DarshitChanpura — DISMISSED** [review]
LGTM!

**Review by DarshitChanpura — COMMENTED** [review]
If you are instead trying to Deprecate the routes itself, you can do something like this:

https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/dlic/rest/api/SecuritySSLCertsApiAction.java#L60

**Review by DarshitChanpura — APPROVED** [review]

File: src/main/java/org/opensearch/security/dlic/rest/api/FlushCacheApiAction.java [hunk 1]
```diff
@@ -39,9 +39,21 @@ public class FlushCacheApiAction extends AbstractApiAction {
     private static final List<Route> routes = addRoutesPrefix(
         ImmutableList.of(
             new Route(Method.DELETE, "/cache"),
-            new Route(Method.GET, "/cache"),
-            new Route(Method.PUT, "/cache"),
-            new Route(Method.POST, "/cache")
+            new DeprecatedRoute(
+                Method.GET,
+                "/cache",
+                "GET is not supported for /cache endpoint and will be removed in the next major version."
```
Comment (cwperks) [inline src/main/java/org/opensearch/security/dlic/rest/api/FlushCacheApiAction.java:45]:
```suggestion
                "GET is not supported for /cache endpoint and will be removed in the next major version. Use DELETE instead."
```

**Review by RyanL1997 — DISMISSED** [review]

File: src/main/java/org/opensearch/security/dlic/rest/api/FlushCacheApiAction.java (hunk 1 above)
Comment (derek-ho) [reply src/main/java/org/opensearch/security/dlic/rest/api/FlushCacheApiAction.java:45]: I don't think we can necessarily assume that they want to DELETE the cache - all we should say is we don't support GET/POST/PUT - wdyt?

Comment (cwperks) [reply src/main/java/org/opensearch/security/dlic/rest/api/FlushCacheApiAction.java:45]: DELETE is the HTTP Verb, so its not deleting the cache, but instructing the nodes to refresh their caches from the security index.

A caller of this API using GET/POST/PUT will already get a not implemented response. I was thinking that the message could be more informative and tell the user the correct http verb to use.

**Review by willyborankin — APPROVED** [review]

File: src/main/java/org/opensearch/security/dlic/rest/api/FlushCacheApiAction.java (hunk 1 above)
Comment (cwperks) [reply src/main/java/org/opensearch/security/dlic/rest/api/FlushCacheApiAction.java:45]: If they are using this API then we can assume the intent is to flush the cache.

Comment (derek-ho) [reply src/main/java/org/opensearch/security/dlic/rest/api/FlushCacheApiAction.java]: Sure makes sense. Added that line to the end of all other routes as well in: [8c99aa0](https://github.com/opensearch-project/security/pull/4776/commits/8c99aa0c07286e28e6628e3d0e814235691a76a6)

**Review by cwperks — APPROVED** [review]

**Review by nibix — APPROVED** [review]

**Pull request merged**